ftplugin/php/splitjoin.vim

```diff
@@ -1,6 +1,6 @@
 let b:splitjoin_split_callbacks = [
       \ 'sj#js#SplitArray',
-      \ 'sj#php#SplitArray',
+      \ 'sj#php#SplitBraces',
```

Comment:
The problem is that the result doesn't make any sense :) This:

``` php
if (
  $foo
) { $a = $bar; }
```

is just not something that I'd want to have as a result.

I remember having a similar discussion on an issue a while back, but I don't think I can find it, so I'll go over my thoughts on this again. Basically, while it makes sense to look around the cursor position for what you want to split, this only makes sense in "equivalent" cases. For instance:

``` php
array(array('foo' => 'bar'), array('bar' => 'baz'))
```

There's no way to determine which of the three arrays to split, unless you respect the cursor position. However, in this case:

``` php
if (array('foo' => 'bar') == array('foo' => 'bar')) { $a = $bar; }
```

It seems clear that the if-clause needs to be split *first*, and then the plugin should even consider the stuff inside the brackets. In the end, splitjoin is a tool to reformat code in order to make it more readable and there are some (I think) clear priorities on how to do that in a lot of cases. I'd say if you're trying to save horizontal space, first you'd split the body, and then, if that's not enough, you'd split the argument list as well.

Or do you prefer to keep the body one-lined even in that case? Does the example at the top fit your coding style? Would you say that it's often that the body of an if-clause is so short that you'd keep it one-lined, while the argument list ends up being too long?